Tests/SonarScanner.MSBuild.PreProcessor.Test/PreProcessorTests.cs

```diff
@@ -242,15 +243,30 @@ public async Task Execute_EndToEnd_SuccessCase_With_Organization()
     public async Task Execute_EndToEnd_UseCli_SuccessCase()
     {
         using var context = new Context(TestContext);
+        context.Factory.ScannerCliResolver.ResolvePath(null).ReturnsForAnyArgs("some/path/to/sonar-scanner");
         var args = new List<string>(CreateArgs()) { "/d:sonar.scanner.useSonarScannerCLI=true" };
-        (await context.Execute(args)).Should().BeTrue();
 
+        (await context.Execute(args)).Should().BeTrue();
         context.AssertDirectoriesCreated();
         context.AssertDownloadMethodsCalled(properties: 1, allLanguages: 1, qualityProfile: 2, rules: 2);
-        context.AssertAnalysisConfig(2);
+        context.AssertAnalysisConfig(2).SonarScannerCliPath.Should().Be("some/path/to/sonar-scanner");
         await context.Factory.EngineResolver.DidNotReceiveWithAnyArgs().ResolvePath(null);
     }
 
+    [TestMethod]
+    public async Task Execute_EndToEnd_EngineNotResolved_FallbackToCli()
```

Comment:
Please add a test with the BuildEnv set to LegacyTFS (if this is the right place for such a test).